### PR TITLE
v1.5.0

### PR DESCRIPTION
**Pull request by melferink**
Addition of SNParray tracks

### CODE REVIEW

**Review by rernst — COMMENTED** [review]
- CytoSNP-850Kv1-2: does not contain MT?
  - .list contains MT reference in header  
- SNParray folder name is not very explanatory. Maybe use BAF_UPD like fingerprint and ED.
- List files contain 'target' while bed files contain an id (rs and other) why not put this on the 'target' location?
- Maybe add a README describing the files and how they were created?

**Review by ellendejong — CHANGES_REQUESTED** [review]
Mainly wondering if you can decrease number of manual/custom bash commands by using existing tools such as Picard or GATK.

File: Tracks/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+```
+cat S31285117_Padded.bed | grep -v track |grep -v browser |sed 's/^chr//g' | awk '{OFS="\t"; print $1,$2,$3,"+", "ssv7_target","ssv7_target","ssv7_target"}' > S31285117_Padded_nochr.bed
+bedtools subtract -a S31285117_Padded_nochr.bed -b ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_100bpflank.bed > S31285117_Padded_nochr_unique.bed
+cat ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_100bpflank.bed S31285117_Padded_nochr_unique.bed | sed 's/^MT/YYYYYYYY/g'| sort -k1,1V -nk2 -nk3 | sed 's/^YYYYYYYY/MT/g'  > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.bed
+cat ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.bed |awk '{OFS="\t" ; print $1,$2+1,$3,$4,$5}' > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_1based.bed
+cat /hpc/diaggen/data/databases/ref_genomes/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.dict ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_1based.bed > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.interval_list
+
+rm S31285117_Padded_nochr.bed
+rm S31285117_Padded_nochr_unique.bed
+rm ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_1based.bed
```
Comment (ellendejong) [inline Tracks/README.md:32]: You can merge these three commands into a single one.

File: Tracks/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## 14-09-2021. Addition of Agilent SureSelect v7 specific targets to the calling target.
+
+Requirements:\
+Original target BED file (i.e. ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_100bpflank.bed ).\
+Track files from Agilent SureSelect v7 elidS31285117 specifically the Padded.ped track. Files can be downloaded from the Agilent website.\
+Dictionary from reference genome (same hg version), i.e. Homo_sapiens.GRCh37.GATK.illumina.dict
+
+```
+cat S31285117_Padded.bed | grep -v track |grep -v browser |sed 's/^chr//g' | awk '{OFS="\t"; print $1,$2,$3,"+", "ssv7_target","ssv7_target","ssv7_target"}' > S31285117_Padded_nochr.bed
```
[the remark this answers is not in the document]
Comment (melferink) [reply Tracks/README.md:24]: Agree, exept the $2+1.  Conversion to 1-based will be later.

File: Tracks/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+## 14-09-2021. Addition of BAIT file for Agilent SureSelect v7 enrichment design elidS31285117.
+
+Requirements:\
+Track files from Agilent SureSelect v7 elidS31285117, specifically the Coverd.bed track. Files can be downloaded from the Agilent website.\
+Dictionary from reference genome (same hg version), i.e. Homo_sapiens.GRCh37.GATK.illumina.dict
+
+```
+cat S31285117_Covered.bed |grep -v browser |grep -v track | sed 's/^chr//g' | awk '{OFS ="\t"; print $1,$2+1,$3,"+","SS_v7_S31285117"}' > S31285117_Covered_nochr_5kol.list
```
[the remark this answers is not in the document]
Comment (melferink) [reply Tracks/README.md:11]: I do not agree with the substr($1,4). It's not always certain if the BED file of Agilent are "chr"  based or not.
Using substr will go wrong if it doesn't contain chr.
sed /^chr// is a saver bet in my opinion.

File: Tracks/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Dictionary from reference genome (same hg version), i.e. Homo_sapiens.GRCh37.GATK.illumina.dict
+
+```
+cat S31285117_Padded.bed | grep -v track |grep -v browser |sed 's/^chr//g' | awk '{OFS="\t"; print $1,$2,$3,"+", "ssv7_target","ssv7_target","ssv7_target"}' > S31285117_Padded_nochr.bed
+bedtools subtract -a S31285117_Padded_nochr.bed -b ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_100bpflank.bed > S31285117_Padded_nochr_unique.bed
+cat ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_100bpflank.bed S31285117_Padded_nochr_unique.bed | sed 's/^MT/YYYYYYYY/g'| sort -k1,1V -nk2 -nk3 | sed 's/^YYYYYYYY/MT/g'  > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.bed
+cat ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.bed |awk '{OFS="\t" ; print $1,$2+1,$3,$4,$5}' > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_1based.bed
+cat /hpc/diaggen/data/databases/ref_genomes/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.dict ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_1based.bed > ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank.interval_list
```
[the remark this answers is not in the document]
Comment (melferink) [reply Tracks/README.md:28]: Agree to use Picard (despite  adding more dependencies). 
However, number of lines do not really decrease as Picard input need some massage as well ;)

**Review by ellendejong — APPROVED** [review]
As discussed separately, I approve these changes with two remarks.

1. In general, I prefer using tools over custom commands to clean-up code.
A specific example in this pull-request is the manual merging of bedfiles which include sorting on chromosomes and the difficulties with MT after X,Y.

1. Secondly, when using a certain file type extension, I think it is good practice to use a format guideline. For example, files within this repos / pull-request are called .bed, and format does not match these guidelines http://genome.ucsc.edu/FAQ/FAQformat#format1 . Ofcourse, this can be solved by either changing the extension to .tsv, or change the content to match.

I will add these two remarks as work items in Azure and hope we can resolve it somewhere in the near future.

**Pull request merged**